intrahospital_api/deployment_check.py

```diff
@@ -0,0 +1,201 @@
+
+It populates a result dict with
+current: The current state of the db.
+batch_load: The result of the batch.
+loadinitial_load: The result of the individual patient loads.
+
+Next to each is a [obs number, lab test number, patient id] which
+do not exist in at least one other load.
+"""
+
+
+import datetime
+from django.conf import settings
+from django.utils import timezone
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from opal.core import serialization
+from opal.models import Patient, Tagging
+from elcid import patient_lists
+from lab import models as lmodels
+from intrahospital_api import models as imodels
+from intrahospital_api import loader
+from intrahospital_api import get_api
+from intrahospital_api import update_lab_tests
+
+
+class RollBackError(Exception):
+    pass
+
+
+def batch_load_patients(since):
+    # this is all done much better after the refactor, but
+    # the below _should_ work ok
+    api = get_api()
+    data_deltas = api.data_deltas(since)
+    loader.update_from_batch(data_deltas)
+
+
+def update_patient(patient):
+    # this is all done much better after the refacot
+    # but the below should work the same as an
+    # initial lab test load
+    api = get_api()
+    hospital_number = patient.demographics_set.first().hospital_number
+    results = api.results_for_hospital_number(hospital_number)
+    update_lab_tests.update_tests(patient, results)
+
+
+def get_patients():
+    """
+    We only look at patients on the bacteraemia list as
+    we these should have lab tests
+    """
+    tags = Tagging.objects.filter(
+        value=patient_lists.Bacteraemia.tag
+    ).filter(
+        archived=False
+    )
+    return Patient.objects.filter(
+        episode__tagging__in=tags
+    )
+
+
+def get_qs(min_dt, max_dt=None):
+    """
+    Return all upstream lab tests for the patients
+    """
+    patients = get_patients()
+    lab_tests = lmodels.LabTest.objects.filter(
+        lab_test_type__istartswith="upstream"
+    ).filter(patient__in=patients)
+
+    max_updated = timezone.make_aware(datetime.datetime.min)
+    if max_dt is None:
+        max_dt = timezone.make_aware(datetime.datetime.max)
+
+    result = {}
+
+    for lab_test in lab_tests:
+        observations = lab_test.extras["observations"]
+        for observation in observations:
+            last_updated_str = observation["last_updated"]
+            if last_updated_str:
+                last_updated = serialization.deserialize_datetime(
+                    last_updated_str
+                )
+
+                if last_updated >= min_dt and last_updated <= max_dt:
+                    result[observation["observation_number"]] = lab_test
+
+                    if max_updated < last_updated:
+                        max_updated = last_updated
+
+    return result, max_updated
+
+
+def get_key(observation_number, lab_test):
+    """
+    A key is the observation number, lab test number and patient id.
```

Comment:
Why don't we just use the phrase observation number instead of inventing this 'key' terminology?
Won't that be a confusing layer of indirection in 3 months?